baremetal/metal3machine_manager_test.go

```diff
@@ -1882,6 +1891,185 @@ var _ = Describe("Metal3Machine manager", func() {
 			Secret:                  newSecret(),
 			ExpectedBMHOnlineStatus: false,
 		}),
+		Entry("NodeReuse enabled, machine is worker, no error expected", testCaseDelete{
+			Host: newBareMetalHost("myhost",
+				&bmh.BareMetalHostSpec{
+					ConsumerRef: consumerRef(),
+					Online:      false,
+				},
+				bmh.StateNone,
+				&bmh.BareMetalHostStatus{
+					Provisioning: bmh.ProvisionStatus{
+						State: bmh.StateNone,
+					},
+				}, false, "metadata", true),
+			Machine: &clusterv1.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "mymachine",
+					Namespace: namespaceName,
+					OwnerReferences: []metav1.OwnerReference{
+						{
+							APIVersion: clusterv1.GroupVersion.String(),
+							Name:       "myMachineSet",
+							UID:        "123456789",
+							Kind:       "MachineSet",
+						},
+					},
+				},
+				Spec: clusterv1.MachineSpec{
+					Bootstrap: clusterv1.Bootstrap{
+						DataSecretName: pointer.StringPtr("mym3machine-user-data")},
+				},
+			},
+			MachineSet: &clusterv1.MachineSet{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: clusterv1.GroupVersion.String(),
+					Kind:       "MachineSet",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "myMachineSet",
+					Namespace: namespaceName,
+					UID:       "123456789",
+					OwnerReferences: []metav1.OwnerReference{
+						{
+							APIVersion: clusterv1.GroupVersion.String(),
+							Kind:       "MachineDeployment",
+							Name:       "test1",
+						},
+					},
+				},
+			},
+			M3Machine: newMetal3Machine("mym3machine", nil, nil, m3mSecretStatus(),
+				&metav1.ObjectMeta{
+					Name:      "mym3machine",
+					Namespace: namespaceName,
+					OwnerReferences: []metav1.OwnerReference{
+						{
+							Kind:       "MachineSet",
+							APIVersion: capm3.GroupVersion.String(),
+							Name:       "test1",
+							UID:        "123456789",
+						},
+						{
+							APIVersion: clusterv1.GroupVersion.String(),
+							Kind:       "Machine",
+							Name:       "test1-bwjsg",
+							UID:        "070f03fb-b0e1-4863-9d95-7fd681d8d257",
+						},
+					},
+					Labels: map[string]string{
+						clusterv1.ClusterLabelName: clusterName,
+					},
+					Annotations: map[string]string{
+						HostAnnotation:                           "baremetalns/myhost",
+						"cluster.x-k8s.io/cloned-from-name":      "abc",
+						"cluster.x-k8s.io/cloned-from-groupkind": capm3.ClonedFromGroupKind,
+					},
+				}),
+			Cluster:             newCluster(clusterName),
+			capm3fasttrack:      "false",
```

Comment:
This can be removed I believe, it should have no effect on this test case

Reply:
Do you mean capm3fasttrack value?

Reply:
Yes

Reply:
removed